src/groupeddataframe/callprocessing.jl

```diff
@@ -90,7 +90,7 @@ end
 function do_call(f::Base.Callable, idx::AbstractVector{<:Integer},
                  starts::AbstractVector{<:Integer}, ends::AbstractVector{<:Integer},
                  gd::GroupedDataFrame, incols::Tuple{AbstractVector}, i::Integer)
-    idx = idx[starts[i]:ends[i]]
+    idx = view(idx, starts[i]:ends[i])
```

Comment:
saves allocations. I left allocating code for more passed columns as I was unsure what is faster - allocate indices or have a doubly nested `view`.

Reply:
I did some more benchmarking and I change the code to use `view` in all cases.

Reply:
I'd think that views based on a range of indices are quite efficient, but only benchmarking can tell.

Reply:
The performance difference is small, but it will be more GC friendly (with a large number of groups it is a bit better to use `view` with a small number of groups it is a bit better to materialize the `idx`).

PR after the change:
```
julia> df = DataFrame(x=rand(1:10^6, 10^7), v1=rand(10^7), v2=rand(10^7));

julia> gdf = groupby(df, :x);

julia> f(v1,v2) = cor(v1, v2)^2
f (generic function with 1 method)

julia> @btime combine($gdf, [:v1, :v2] => f => :r2);
  932.859 ms (1000270 allocations: 179.30 MiB)

julia> df = DataFrame(x=rand(1:10, 10^7), v1=rand(10^7), v2=rand(10^7));

julia> gdf = groupby(df, :x);

julia> f(v1,v2) = cor(v1, v2)^2
f (generic function with 1 method)

julia> @btime combine($gdf, [:v1, :v2] => f => :r2);
  287.085 ms (321 allocations: 76.31 MiB)
```

current release:
```
julia> df = DataFrame(x=rand(1:10^6, 10^7), v1=rand(10^7), v2=rand(10^7));

julia> gdf = groupby(df, :x);

julia> f(v1,v2) = cor(v1, v2)^2
f (generic function with 1 method)

julia> @btime combine($gdf, [:v1, :v2] => f => :r2);
  810.070 ms (315 allocations: 22.91 MiB)

julia> df = DataFrame(x=rand(1:10, 10^7), v1=rand(10^7), v2=rand(10^7));

julia> gdf = groupby(df, :x);

julia> f(v1,v2) = cor(v1, v2)^2
f (generic function with 1 method)

julia> @btime combine($gdf, [:v1, :v2] => f => :r2);
  269.337 ms (302 allocations: 18.81 KiB)
```